src/hotspot/share/nmt/virtualMemoryTracker.hpp

```diff
@@ -303,6 +303,9 @@ class ReservedMemoryRegion : public VirtualMemoryRegion {
   ReservedMemoryRegion(address base, size_t size) :
     VirtualMemoryRegion(base, size), _stack(NativeCallStack::empty_stack()), _flag(mtNone) { }
 
+  ReservedMemoryRegion(address base, size_t size, MEMFLAGS flag) :
+    VirtualMemoryRegion(base, size), _stack(NativeCallStack::empty_stack()), _flag(flag) { }
```

Comment:
The function above uses mtNone. I find that a bit dubious, but I understand that it is done to be able to write code like this:
```
    ReservedMemoryRegion* rmr = VirtualMemoryTracker::_reserved_regions->find(ReservedMemoryRegion(addr, size));
```

Unfortunately, it opens up the door for people to accidentally use that version instead of this new version that you have written. Could we get rid of the version using mtNone somehow?

The same question goes for the version above that, which has a "MEMFLAGS flag = mtNone". (GH doesn't allow me to comment on lines that you haven't changed)

Reply:
`mtNone` as default value is no longer valid.